pkg/controllers/networkpolicy/networkpolicy.go

```diff
@@ -426,9 +427,6 @@ func (s *securityGroupImpl) deleteImpl(c cloudSecurityGroup, membershipOnly bool
 	ch := securitygroup.CloudSecurityGroup.DeleteSecurityGroup(&s.id, membershipOnly)
 	go func() {
 		err := <-ch
-		if err == nil && !membershipOnly {
-			s.deleteSgRulesFromIndexer(r)
```

Comment:
why you dont want to remove the rules from indexer for success case?

Reply:
Discussed offline and updated PR.